source/extensions/filters/http/ext_proc/config.cc

```diff
@@ -11,11 +12,19 @@ namespace ExternalProcessing {
 
 Http::FilterFactoryCb ExternalProcessingFilterConfig::createFilterFactoryFromProtoTyped(
     const envoy::extensions::filters::http::ext_proc::v3alpha::ExternalProcessor& proto_config,
-    const std::string&, Server::Configuration::FactoryContext&) {
-  const auto filter_config = std::make_shared<FilterConfig>(proto_config);
+    const std::string&, Server::Configuration::FactoryContext& context) {
+  const uint32_t timeout_ms =
+      PROTOBUF_GET_MS_OR_DEFAULT(proto_config.grpc_service(), timeout, DefaultTimeout);
```

Comment:
It's passed to the gRPC client when starting a stream in the start of decodeHeaders.

We also need a timeout in case we send a message on the stream and get no response from the remote server in time, but that's not implemented. I think that should be some multiple of the generic gRPC timeout used here. Don't know if we have a good pattern for that elsewhere in Envoy that we can copy.

Reply:
@yuval-k what's the latest story for `ext_authz` and these timeouts?

Reply:
with ext-auth it ended-up not being a problem (I reverted my change), as the async client sends the headers on body at the same time for a single request (I think the source of my confusion was that the request was outright rejected, not pending).

Reply:
Thanks @yuval-k. It's actually a bit strange to use this timeout for long-lived streaming `ext_proc`, since it's not a request timeout but rather an initial frame on stream timeout. I wonder if it makes sense to better document this in the API at least.

Reply:
LMK if you think I should use a different timeout -- the async client does indeed take a timeout as a parameter so I figured it'd make sense to give it one.

Reply:
I'd like to leave the timeout the way it is for this PR and introduce a more complete timeout mechanism in a future PR.